components/kyma-environment-broker/internal/avs/client.go

```diff
@@ -12,12 +12,12 @@ import (
 
 type clientHolder struct {
 	httpClient        *http.Client
-	avsConfig         Config
+	avsConfig         *Config
```

Comment:
why did you change this to a pointer?

Reply:
As described in commit details: 
Switches to pointer-type avs configs across KEB, as in the future the configuration might get updated at runtime (provisioning, deprovisoning) and the changes should be properly delegated.

Reply:
The configuration is set by environment variables. There is no any runtime configuration change right now. The change to a pointer is not enough. What about synchronization? you are sharing data without any locking? is it safe? Let's prepare for a runtime config change if there is an issue for that.

Reply:
I agree that this functionality is currently underused, as there are no runtime config changes. However, this allows a more robust Evaluation Assistant class-specific configuration. In this case, we wouldn't have to rely on the singular classes but use a more OOP approach (possibility of not only using two EvalAssistants). The runtime changes should then be reflected on all the inherited EvalAssistants, which this is trying to accomplish. To ensure consistency, mutex implementation would be required.
Regarding safety, there is no way of making read-only objects or creating constant pointer passes in Go, hence any implementation using this approach is by its nature -- unsafe. We can switch from using current avs config object to a getter-setter model, rendering the object changes safer, however, this might get a bit longer in case of object extensions.

In any case, thanks for pointing out the possible twisters. What do you think would be the best approach to follow, extending this further or falling back to static config objects?

Reply:
I'd like to see a use case of change the config in the runtime before the implementation. Let's do not implement something which is not defined.
Passing a struct as a value makes a copy which prevents runtime changes. It is something like passing value object and it is ok.